sdk/trace/provider.go

```diff
@@ -109,32 +109,48 @@ func (p *Provider) Tracer(name string, opts ...apitrace.TracerOption) apitrace.T
 func (p *Provider) RegisterSpanProcessor(s SpanProcessor) {
```

Comment:
Can we add a test to check for this FIFO processing behaviour?

Reply:
Sure, will add a test.

Reply:
Hi, @MrAlias , at first glance, I though this should be straightforward by defining an array of processors, add them to the provider, and check the designed order. However, after I register the spanprocessors and want to retrieve them from the provider like 

```go
gotSps := p.spanProcessors.Load().(spanProcessorStates)
```
I realized that the current trace provider does not export the processors. A hack would be to add a method to return the registered processors, but I'd like to check with you to see if I miss anything. Thanks.

Reply:
I was envisioning a more integration type test here where two span processors would be registered that each modified some state held by the test function. After a span was started and ended the state could be verified that it was updated in the correct order.

That said this comment wasn't a blocker for this issue. If you want @huikang we can create an issue to track this and merge this PR. Would that work?

Reply:
Sounds a much better idea. 

When you have a chance to create an issue to track the test scenario, I am happy to take that. Thanks.

Reply:
Created https://github.com/open-telemetry/opentelemetry-go/issues/1203 to track.